proto/trailers.proto

```diff
@@ -0,0 +1,34 @@
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/momentohq/client-sdk-go;client_sdk_go";
+option java_package = "momento.trailers";
+option csharp_namespace = "Momento.Protos.Trailers";
+
+package trailers;
+
+// Found on the `retry_disposition` trailer for error responses, when the value is known by the service
+enum RetryDisposition {
+     // This rpc is safe to retry, even if it is non-idempotent. It was not executed by the server.
+    retryable = 0;
+    // This rpc may be safe to retry, but it may have been applied.
+    // Non-idempotent commands should not be retried, unless configured explicitly.
+    // Idempotent commands should be considered eligible for retry.
+    possibly_retryable = 1;
+}
+
+// Found on the `err` trailer for error responses, when the error is sent by the server and it has a relevant value to set
+enum Err {
```

Comment:
I think I like this idea but it will be a breaking change until the SDKs are updated, right?

Reply:
Given that we are using these as string's and not the binary protobuf format, yes, changing a metadata name server side would be breaking for SDKs. If that's what you mean, I may be lacking some sdk insight.

Reply:
> not the binary protobuf format

Wait, I'm confused. If we're not going to use the binary protobuf format then why are we defining these as binary protos?

Since you are introducing these protos in this PR I was assuming that meant that you were going to start using them as the actual values for the `err` metadata. Since the SDKs are currently expecting strings for those, then, as I assumed this PR implies we will be changing from strings to binary protos, then the SDKs would need to be updated and there would be a period of time in between where they were broken.

If you are still planning to send strings back for the `err` metadata then I'm a little confused as to what this proto is for.

Reply:
@ben-kugler if it would be easier to get aligned on this with a short call with a few folks, happy to do that!

Reply:
Yeah! That would be nice, I'll get that going. Also, some extra context, the initial idea was to just include trailer warnings as comments, when doing this I then took it a step further to add them as a proto. I do think we should take it further and use the protobuf format if we are going to include them as proto's as you've pointed out. Not doing so would be confusing. If we decide that we aren't going to go in that direction, maybe removing the `trailers` proto would be best.

Reply:
cool. I think it'd be great to have @malandis, @kvcache and @virratanasangpunth in that meeting if possible as they've all been involved in earlier rounds of convo about this.